lib/Dialect/Torch/Transforms/VerifyConversionToValueSemantics.cpp

```diff
@@ -0,0 +1,61 @@
+//===- VerifyConversionToValueSemantics.cpp ----------------------*- C++-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+
+#include "mlir/IR/BuiltinOps.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchTypes.h"
+#include "torch-mlir/Dialect/Torch/Transforms/Passes.h"
+
+using namespace mlir;
+using namespace mlir::torch::Torch;
+
+static LogicalResult checkValueType(Operation *op, Value value) {
+  auto isNotValueTensorType = value.getType().isa<NonValueTensorType>();
+  return isNotValueTensorType ? op->emitError("found non-value tensor type")
```

Comment:
Can you expand this error message a bit, since it will be what most devs run into? I would recommend adding a note saying "this is likely due to a missing case in the MaximizeValueSemantics pass"